### ISSUE TITLE
[BUG] - It is possible to get duplicate ingredient_units

### ISSUE THREAD

**Issue by johnsonm** [mealie-recipes/mealie#2510]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/)

### What is the issue you are experiencing?

I don't know how to reproduce this, either on the demo site or on a new install, but:

```
--
-- Data for Name: ingredient_units; Type: TABLE DATA; Schema: public; Owner: mealie
--

COPY public.ingredient_units (created_at, update_at, id, group_id, name, description, abbreviation, fraction, use_abbreviation) FROM stdin;
...
2023-08-19 16:26:04.419197      2023-08-19 16:26:04.419202      aa67e5a2-91bf-4520-be2f-3689647f1f63    a3f0dcc1-d69f-411a-9bb8-7202385bf253    gram                    t       f
2023-08-19 16:26:06.643603      2023-08-19 16:26:06.643608      6adfa8fb-1d0d-45db-8ea3-c4152172d68e    a3f0dcc1-d69f-411a-9bb8-7202385bf253    gram                    t       f
```

My wife and I were both entering recipes together and may have entered this at the same time but doing that just two seconds apart seems unlikely.

I don't see a way in the UI to unify or delete units. I had to use psql to delete the one that wasn't referenced anywhere else from the database (with `delete from ingredient_units where id='6adfa8fb-1d0d-45db-8ea3-c4152172d68e';`). (Similarly, I had to do the same thing for a unit with a typo; I fixed the reference in the recipe and had no references left to it, but it would keep showing up in the UI until I removed it from the database.)

### Steps to Reproduce

I wish I knew. 

### Please provide relevent logs

None obvious in meaile.log

### Mealie Version

Version
v1.0.0beta-5
Build
ee121a12f8db33ecb4db5f8582f7ea9788d019e4 

### Deployment

Docker (Linux)

### Additional Deployment Details

Postgres backend, configured following documentation.

**Comment by regularguy01** [comment]
What happens if you Follow the Seed Process? https://nightly.mealie.io/documentation/getting-started/faq/#how-do-i-enable-smart-ingredient-handling

**Comment by regularguy01** [comment]
I checked my Ingredients_units table in psql I only have one gram and no duplicates.

**Comment by JvdMaat** [comment]
I seeded mine initially (not using postgres BTW, just the built-in DB), and last week found I had about two dozen duplicate units (Also no idea how/why). I went into group data management (https://mealie/group/data/units) and removed the duplicates. But then I had to go through all my recipes and update the ones that were using the duplicate value I removed.
I think for foods, units, and labels, we should check for duplicates and not allow them. (Just like duplicate recipes are not allowed). Since it makes no sense to have duplicates there.
And apparently I overlooked kilogram as that one is still double.

**Comment by regularguy01** [comment]
> I seeded mine initially (not using postgres BTW, just the built-in DB)


I have Postgres.  Wonder if its a SQLite or SQLite to Postgres thing.  

Unrelated I can make duplicate recipes by using the same url twice.

**Comment by johnsonm** [comment]
I didn't see the documentation for seeding and have done it all by hand. I have lots already added already... I'm just getting started.

I have been using Postgres since the beginning.

But the seeding page showed me the "Data Management" I hadn't yet noticed, and I've been combining duplicates, renaming one of "foo" to "foo2" and then using a source of "foo2" and a target of "foo" to merge them together.

The worst so far is five of "orange zest". I think there were places in the UI where it wasn't responsive enough to show that an ingredient had been added and so we added it again.
